modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx

```diff
@@ -68,16 +76,54 @@ export default function Fields({
 
 	useEffect(() => {
 		const makeFetch = async () => {
-			const objectDefinition = await API.getObjectDefinitionByExternalReferenceCode(
+			setLoad(true);
+
+			const objectDefinitionResponse = await API.getObjectDefinitionByExternalReferenceCode(
 				objectDefinitionExternalReferenceCode
 			);
 
-			setCreationLanguageId(objectDefinition.defaultLanguageId);
+			const url = createResourceURL(baseResourceURL, {
+				objectDefinitionId: objectDefinitionResponse.id,
+				p_p_resource_id: '/object_definitions/get_object_field_types',
+			}).href;
+
+			const objectFieldTypesResponse = await fetch(url, {
+				method: 'GET',
+			});
+
+			const {
+				objectFieldTypes: newObjectFieldTypes,
+			} = (await objectFieldTypesResponse.json()) as {
+				objectFieldTypes: ObjectFieldType[];
+			};
+
+			setCreationLanguageId(objectDefinitionResponse.defaultLanguageId);
+			setObjectFieldTypes(newObjectFieldTypes);
+
+			setTimeout(() => {
+				setLoad(false);
+			}, 200);
 		};
 
 		makeFetch();
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [objectDefinitionExternalReferenceCode]);
 
+	function objectFieldBusinessTypeDataRenderer({
+		itemData,
+	}: {
+		itemData: ItemData;
```

Comment:
@igor-franca ItemData is actually ObjectField, we can create a new interface to replace all usage of this ItemData interface

```tsx
interface ObjectFieldItemData {
	itemData: ObjectField;
}
```
And replace all other functions that use it with something like:

```suggestion
: ObjectFieldItemData) 
```